vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt

```diff
@@ -270,139 +271,103 @@ class OidcSiopWallet(
-
-                    if (!dnsNames.contains(params.parameters.clientId))
-                        return KmmResult.failure<AuthenticationResponseParameters>(OAuth2Exception(Errors.INVALID_REQUEST))
-                            .also { Napier.w("client_id_scheme is $clientIdScheme, but client_id does not match any dnsName in the leaf certificate") }
-
-                    if (!((params.parameters.responseMode == DIRECT_POST) ||
-                                (params.parameters.responseMode == DIRECT_POST_JWT))
-                    ) {
-                        val parsedUrl =
-                            params.parameters.redirectUrl?.let { Url(it) }
-                                ?: return KmmResult.failure<AuthenticationResponseParameters>(OAuth2Exception(Errors.INVALID_REQUEST))
-                                    .also { Napier.w("client_id_scheme is $clientIdScheme, but no redirect_url was provided") }
-                        //TODO  If the Wallet can establish trust in the Client Identifier authenticated through the certificate it may allow the client to freely choose the redirect_uri value
-                        if (parsedUrl.host != params.parameters.clientId) return KmmResult.failure<AuthenticationResponseParameters>(
-                            OAuth2Exception(Errors.INVALID_REQUEST)
-                        ).also { Napier.w("client_id_scheme is $clientIdScheme, but no redirect_url was provided") }
-                    }
-                } else {
-                    val uris = leaf.tbsCertificate.subjectAlternativeNames?.uris
-                        ?: return KmmResult.failure<AuthenticationResponseParameters>(OAuth2Exception(Errors.INVALID_REQUEST))
-                            .also { Napier.w("client_id_scheme is $clientIdScheme, but no URIs were found in the leaf certificate") }
-                    if (!uris.contains(params.parameters.clientId))
-                        return KmmResult.failure<AuthenticationResponseParameters>(OAuth2Exception(Errors.INVALID_REQUEST))
-                            .also { Napier.w("client_id_scheme is $clientIdScheme, but client_id does not match any URIs in the leaf certificate") }
-
-                    if (params.parameters.clientId != params.parameters.redirectUrl)
-                        return KmmResult.failure<AuthenticationResponseParameters>(OAuth2Exception(Errors.INVALID_REQUEST))
-                            .also { Napier.w("client_id_scheme is $clientIdScheme, but client_id does not match redirect_uri") }
-                }
-            }
+        if (clientIdScheme.isAnyX509()) {
+            runCatching { params.verifyClientIdSchemeX509() }
+                .onFailure { return KmmResult.failure(it) }
         }
 
         // params.clientIdScheme is assumed to be OpenIdConstants.ClientIdSchemes.REDIRECT_URI,
         // because we'll require clientMetadata to be present, below
```

Comment:
Probably a good time to also remove this comment

Reply:
Done in https://github.com/a-sit-plus/kmm-vc-library/pull/68/commits/5cda85bcada80db90bdd352ba0092b1c3c0553a9